doc/guides/device_mgmt/ota.rst

```diff
@@ -0,0 +1,65 @@
+.. _ota:
+
+Over-the-Air Update
+###################
+
+Overview
+********
+
+Over-the-Air (OTA) Update is a method for delivering firmware updates to remote
+devices using a network connection. Although the name implies a wireless
+connection, devices receiving updates over a wired connection (such as Ethernet)
+are still commonly referred to by OTA. This approach requires server
+infrastructure to host the firmware binary and implement a method of signaling
+when an update is available.
+
+The :ref:`dfu` section discusses upgrading Zephyr firmware using MCUboot. The
+same method can be used as part of OTA. The binary is first downloaded
+into an unoccupied code partition, then flashed using the `MCUboot bootloader`_
+process.
+
+Examples of OTA
+***************
+
+Golioth
+=======
+
+`Golioth`_ is an IoT management platform that includes Zephyr OTA updates.
+Devices are configured to observe your available firmware revisions on the
+Golioth Cloud. When a new version is available, the device downloads and flashes
+the binary.
+
+Security is a concern with OTA updates. In this implementation, the connection
+between cloud and device is secured using TLS/DTLS, and the signed firmware binary
+is confirmed by MCUboot before flashing occurs.
+
+1. A working sample can be found on the `Golioth Zephyr-SDK repository`_
+2. The `Golioth OTA documentation`_ includes complete information about the
+   versioning process
+
+Eclipse hawkBit |trade|
+=======================
+
+`Eclipse hawkBit`_ |trade| is a domain independent back-end framework for rolling out
+software updates to constrained edge devices as well as more powerful
+controllers and gateways connected to IP based networking infrastructure.
+
+There is a :ref:`hawkbit-api-sample` included in the
+Zephyr :ref:`mgmt-samples` section.
+
+UpdateHub
+=========
+
+`UpdateHub`_ is an enterprise-grade solution which makes it simple to remotely
+update all your Linux-based devices in the field, with maximum security and
+efficiency, while you focus on adding value to your product.
+
+There is an :ref:`updatehub_fota_sample` included in the Zephyr
+:ref:`mgmt-samples` section.
```

Comment:
Were lwm2m firmware updates via the standard Firmware object (see `CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT`) intentionally omitted? If so, why?

Reply:
It was omitted because I didn't know about it. It seems the firmware upgrade support in LwM2M is largely undocumented (in the Zephyr docs) and not included in the LwM2M client sample. I'm adding a section to reference it as an option.